cmd/apiserver/main.go

```diff
@@ -290,23 +320,38 @@ func main() {
 					BrowserFlow:     webAuth,
 					DeviceFlow:      cliAuth,
 					Store:           store,
-					RedisServer:     cCtx.String("redis-server"),
-					RedisDB:         cCtx.Int("redis-db"),
+					SessionStore:    sessionStore,
 				})
 				if err != nil {
 					log.Fatal(err)
 				}
 
-				server := &http.Server{
+				httpServer := &http.Server{
 					Addr:              cCtx.String("listen"),
 					Handler:           router,
 					ReadTimeout:       5 * time.Second,
 					ReadHeaderTimeout: 5 * time.Second,
 					WriteTimeout:      10 * time.Second,
 				}
+				defer util.IgnoreError(httpServer.Close)
 
 				go func() {
-					if err = server.ListenAndServe(); err != nil {
+					if err = httpServer.ListenAndServe(); err != nil {
+						log.Fatal(err)
```

Comment:
I know we're already `log.Fatal` ing, but could we not do that please? especially from a goroutine.
it's going to immediately os.Exit(1) and deferred functions aren't going to get called.
we should probably propagate the error back to the main func on a channel then exit from there.

Reply:
I can fix that.